modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/service/impl/ListTypeEntryLocalServiceImpl.java

```diff
@@ -119,6 +126,10 @@ public ListTypeEntry updateListTypeEntry(
 		ListTypeEntry listTypeEntry = listTypeEntryPersistence.findByPrimaryKey(
 			listTypeEntryId);
 
+		if (Validator.isNotNull(externalReferenceCode)) {
```

Comment:
I think the update method should follow the same behavior, if the value is null, set the key value.
See example [ObjectEntryLocalServiceImpl#_setExternalReferenceCode](https://github.com/liferay/liferay-portal/blob/b69bd6a04fa27629d64448389d92bf7743e4b008/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java#L2939)